SmartDeviceLink/public/SDLVoiceCommand.m

```diff
@@ -22,12 +23,21 @@ - (instancetype)initWithVoiceCommands:(NSArray<NSString *> *)voiceCommands handl
     self = [super init];
     if (!self) { return nil; }
 
-    _voiceCommands = voiceCommands;
+    self.voiceCommands = voiceCommands;
     _handler = handler;
 
     return self;
 }
 
+- (void)setVoiceCommands:(NSArray<NSString *> *)voiceCommands {
+    if (voiceCommands.count > 0) {
+        NSSet *voiceCommandsSet = [[NSSet alloc] initWithArray:voiceCommands];
+        _voiceCommands = [voiceCommandsSet allObjects];
+    } else {
+        _voiceCommands = voiceCommands;
+    }
+}
```

Comment:
Because this is publicly not settable, I don't think it's best to create a setter and a private readwrite property. Instead, make a class method called `sdl_removeDuplicateStrings:`. Then in the initializer, you can do `_voiceCommands = [self.class sdl_removeDuplicateStrings:voiceCommands];`